filters/include/pcl/filters/impl/box_clipper3D.hpp

```diff
@@ -108,21 +108,23 @@ pcl::BoxClipper3D<PointT>::transformPoint (const PointT& pointIn, PointT& pointO
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // ToDo use product on point.getVector3fMap () and transformatio_.col (i) to use the SSE advantages of eigen
+// http://www.opengl-tutorial.org/beginners-tutorials/tutorial-3-matrices/#an-introduction-to-matrices
 template<typename PointT> bool
 pcl::BoxClipper3D<PointT>::clipPoint3D (const PointT& point) const
 {
-  return  (fabs(transformation_.data () [ 0] * point.x +
-                transformation_.data () [ 3] * point.y +
-                transformation_.data () [ 6] * point.z +
-                transformation_.data () [ 9]) <= 1 &&
-           fabs(transformation_.data () [ 1] * point.x +
-                transformation_.data () [ 4] * point.y +
-                transformation_.data () [ 7] * point.z +
-                transformation_.data () [10]) <= 1 &&
-           fabs(transformation_.data () [ 2] * point.x +
-                transformation_.data () [ 5] * point.y +
-                transformation_.data () [ 8] * point.z +
-                transformation_.data () [11]) <= 1 );
+  Eigen::Matrix4f mat = transformation_.matrix(); 
+  return  (fabs(mat(0,0) * point.x +
+                mat(0,1) * point.y +
+                mat(0,2) * point.z +
+                mat(0,3)) <= 1 &&
+           fabs(mat(1,0) * point.x +
+                mat(1,1) * point.y +
+                mat(1,2) * point.z +
+                mat(1,3)) <= 1 &&
+           fabs(mat(2,0) * point.x +
+                mat(2,1) * point.y +
+                mat(2,2) * point.z +
+                mat(2,3)) <= 1 );
```

Comment:
Replace this entire block for 
```cpp
Vector4fMapConst pt = point.getVector4fMap ();
return (std::abs (mat.row (0) * pt) <= 1 && std::abs (mat.row (1) * pt) <= 1 && std::abs (mat.row (2) * pt) <= 1);
```
You can make multiline if for easier readability.

Note: I'm assuming point.data[3] to be equal to 1, as it is normally initialized.